src/components/OnePost.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { Post } from '../types/Post';
+
+type Props = {
+  post: Post;
+  activePostId: number | null;
+  setActivePostId: (param: number | null) => void;
+};
+
+export const OnePost: React.FC<Props> = ({ post, activePostId, setActivePostId }) => {
```

Comment:
The component `OnePost` is correctly implemented with the necessary props and logic to handle post selection. However, ensure that the `Post` type imported from '../types/Post' includes all the necessary fields used in this component.